DatadogSessionReplay/Sources/Writers/ResourcesWriter.swift

```diff
@@ -16,25 +16,99 @@ internal protocol ResourcesWriting {
 
 internal class ResourcesWriter: ResourcesWriting {
     /// An instance of SDK core the SR feature is registered to.
-    private weak var core: DatadogCoreProtocol?
+    private let scope: FeatureScope?
+    private let telemetry: Telemetry
+
+    @ReadWriteLock
+    private var knownIdentifiers = Set<String>() {
+        didSet {
+            if let knownIdentifiers = knownIdentifiers.asData() {
+                scope?.dataStore.setValue(
+                    knownIdentifiers,
+                    forKey: Constants.knownResourcesKey
+                )
+            }
+        }
+    }
 
     init(
-        core: DatadogCoreProtocol
+        core: DatadogCoreProtocol,
+        dataStoreResetTime: TimeInterval = TimeInterval(30).days
```

Comment:
It's minimal effort on our side. Lowest possible SR retention is 30 days. With this, we enable gradual removal of unused image resources. Meaning that if backend clears stored images on their end every 30 days, we'll make sure to reupload only images that we see during the replay.